src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -219,8 +219,11 @@ function VisEditor($rootScope, $scope, $route, timefilter, AppState, $window, kb
       return savedVis.lastSavedTitle || `${savedVis.title} (unsaved)`;
     };
 
-    $scope.$watch('searchSource.get("index").timeFieldName', function (timeField) {
-      timefilter.enabled = !!timeField;
+    $scope.$watchMulti([
+      'searchSource.get("index").timeFieldName',
+      'vis.type.requiresTimePicker',
+    ], function ([timeField, requiresTimePicker]) {
+      timefilter.enabled = Boolean(timeField || requiresTimePicker);
```

Comment:
should we make it possible for visualization to also specifically opt out of time picker ? so timeField rule only applies if requiresTimePicker is undefined ?